lookout/style/format/benchmarks/general_report.py

```diff
@@ -72,34 +44,47 @@ def _load_jinja2_template(report_template_filename: str) -> jinja2.Template:
     return template
 
 
-def generate_quality_report(language: str, ptr: ReferencePointer, vnodes, max_files: int):
+def generate_quality_report(language: str, ptr: ReferencePointer, vnodes: Sequence[VirtualNode],
+                            y: numpy.ndarray, y_pred_pure: numpy.ndarray, max_files: int,
+                            target_names: Sequence[str]):
     """Generate report: classification report, confusion matrix, files with most errors."""
-    y, y_pred, vnodes_y, target_names = convert_to_sklearn_format(vnodes)
     # classification report
-    c_report = classification_report(y, y_pred, output_dict=True, target_names=target_names,
-                                     labels=list(range(len(target_names))))
+    have_prediction = y_pred_pure >= 0
+    # Predicted Positive Condition Rate calculation
+    ppcr = numpy.sum(have_prediction) / have_prediction.shape[0]
+
+    c_report = classification_report(
+        y[have_prediction], y_pred_pure[have_prediction], output_dict=True,
+        target_names=target_names, labels=list(range(len(target_names))))
+    c_report_full = classification_report(
+        y, y_pred_pure, output_dict=True,
+        target_names=target_names, labels=list(range(len(target_names))))
+
     avr_keys = ["macro avg", "micro avg", "weighted avg"]
     c_sorted = OrderedDict((key, c_report[key])
                            for key in sorted(c_report, key=lambda k: -c_report[k]["support"])
                            if key not in avr_keys)
     for key in avr_keys:
         c_sorted[key] = c_report[key]
     # confusion matrix
-    mat = confusion_matrix(y, y_pred)
+    mat = confusion_matrix(y, y_pred_pure)
     # sort files by mispredictions
     file_mispred = []
-    for gt, pred, vn in zip(y, y_pred, vnodes_y):
-        if gt != pred:
-            file_mispred.append(vn.path)
+    for vnode in vnodes:
+        if hasattr(vnode, "y_old") and vnode.y != vnode.y_old:
+            file_mispred.append(vnode.path)
     file_stat = Counter(file_mispred)
     to_show = file_stat.most_common()
     if max_files > 0:
         to_show = to_show[:max_files]
+    json_report = dict(cl_report=c_sorted,
+                       cl_report_full=c_report_full,
+                       ppcr=ppcr)
 
     template = _load_jinja2_template("quality_report.md.jinja2")
     # TODO(vmarkovtsev): move all the logic inside the template
-    res = template.render(language=language, ptr=ptr, cl_report=c_sorted, conf_mat=mat,
-                          target_names=target_names, files=to_show)
+    res = template.render(language=language, ptr=ptr, conf_mat=mat, target_names=target_names,
```

Comment:
No, this is no good.

1. No `json_report`. Write it inside.
2. No `**kwargs`, you should be explicit in this case.